commit 0769b2ce29beb99fcaf174aed69968c7b9fac1f1
Author: Benjamin Gilbert <bgilbert@backtick.net>
Date:   Sun Sep 14 15:48:17 2014 -0400

    If EC2 max_spot_price is exceeded, limit bid rather than bailing
    
    http://trac.buildbot.net/ticket/2898

diff --git a/master/buildbot/buildslave/ec2.py b/master/buildbot/buildslave/ec2.py
index 64abf29..a204db8 100644
--- a/master/buildbot/buildslave/ec2.py
+++ b/master/buildbot/buildslave/ec2.py
@@ -348,15 +348,9 @@ class EC2LatentBuildSlave(AbstractLatentBuildSlave):
             target_price = 0.02
         else:
             target_price = (price_sum / price_count) * self.price_multiplier
-        if target_price > self.max_spot_price:
-            log.msg('%s %s calculated spot price %0.2f exceeds '
-                    'configured maximum of %0.2f' %
-                    (self.__class__.__name__, self.slavename,
-                     target_price, self.max_spot_price))
-            raise interfaces.LatentBuildSlaveFailedToSubstantiate()
-        else:
-            log.msg('%s %s requesting spot instance with price %0.2f.' %
-                    (self.__class__.__name__, self.slavename, target_price))
+        target_price = min(target_price, self.max_spot_price)
+        log.msg('%s %s requesting spot instance with price %0.2f.' %
+                (self.__class__.__name__, self.slavename, target_price))
         reservations = self.conn.request_spot_instances(
             target_price, self.ami, key_name=self.keypair_name,
             security_groups=[
